hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Event.java

```diff
@@ -98,23 +102,50 @@ public Event() {
     }
 
     public Event(String tenantId, String id, String category, String text) {
-        this(tenantId, id, category, text, null, null);
+        this(tenantId, id, System.currentTimeMillis(), null, category, text, null, null);
+    }
+
+    public Event(String tenantId, String id, String dataId, String category, String text) {
+        this(tenantId, id, System.currentTimeMillis(), dataId, category, text, null, null);
     }
 
     public Event(String tenantId, String id, String category, String text, Map<String, String> context) {
-        this(tenantId, id, category, text, context, null);
+        this(tenantId, id, System.currentTimeMillis(), null, category, text, context, null);
+    }
+
+    public Event(String tenantId, String id, String dataId, String category, String text, Map<String, String> context) {
+        this(tenantId, id, System.currentTimeMillis(), dataId, category, text, context, null);
+    }
+
+    public Event(String tenantId, String id, long ctime, String category, String text) {
+        this(tenantId, id, ctime, null, category, text, null, null);
+    }
+
+    public Event(String tenantId, String id, long ctime, String dataId, String category, String text) {
+        this(tenantId, id, ctime, dataId, category, text, null, null);
+    }
+
+    public Event(String tenantId, String id, long ctime, String category, String text,
+                 Map<String, String> context) {
+        this(tenantId, id, ctime, null, category, text, context, null);
+    }
+
+    public Event(String tenantId, String id, long ctime, String dataId, String category, String text,
+                 Map<String, String> context) {
+        this(tenantId, id, ctime, dataId, category, text, context, null);
     }
 
-    public Event(String tenantId, String id, String category, String text, Map<String, String> context,
-            Map<String, String> tags) {
+    public Event(String tenantId, String id, long ctime, String dataId, String category, String text,
+                 Map<String, String> context, Map<String, String> tags) {
         this.tenantId = tenantId;
         this.id = id;
+        this.ctime = ctime;
```

Comment:
I think we should treat this like we do in Data, and assign currentTime if ctime <= 0.  Don't let bogus ctimes be passed in.